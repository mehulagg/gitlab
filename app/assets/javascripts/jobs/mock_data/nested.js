export default {
  lines: [
    {
      offset: 0,
      content: [{ text: 'Hello' }],
    },
    {
      offset: 5,
      content: [{ text: 'foo' }],
      sections: ['prepare-script'],
      section_header: true,
    },
    {
      offset: 52,
      content: [{ text: 'bar' }],
      sections: ['prepare-script', 'prepare-script-nested'],
      section_header: true,
      section_duration: "00:02",
    },
    {
      offset: 80,
      content: [{ text: 'this is a collapsible nested section' }],
      sections: ['prepare-script', 'prepare-script-nested'],
    },
    {
      offset: 106,
      content: [],
      sections: ['prepare-script'],
      section_duration: "00:03",
    },
    {
      offset: 155,
      content: [],
    },
  ],
};
