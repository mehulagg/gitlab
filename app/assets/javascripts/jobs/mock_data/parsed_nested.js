export default {
  lines: [
    {
      offset: 0,
      content: [{ text: 'Hello' }],
      lineNumber: 0,
    },
    {
      offset: 5,
      section_header: true,
      isHeader: true,
      isClosed: true,
      line: {
        content: [{ text: 'foo' }],
        sections: ['prepare-script'],
        lineNumber: 1,
      },
      section_duration: "00:03",
      lines: [
        {
          section_header: true,
          section_duration: "00:02",
          isHeader: true,
          isClosed: true,
          line: {
            offset: 52,
            content: [{ text: 'bar' }],
            sections: ['prepare-script', 'prepare-script-nested'],
            lineNumber: 2,
          },
          lines: [
            {
              offset: 80,
              content: [{ text: 'this is a collapsible nested section' }],
              sections: ['prepare-script', 'prepare-script-nested'],
              lineNumber: 3.
            }
          ]
        }
      ]
    },
  ],
};
