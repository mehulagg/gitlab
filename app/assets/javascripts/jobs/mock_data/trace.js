// Using https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/278912562

export default {
  id: 278912562,
  status: 'failed',
  complete: true,
  state:
    'eyJvZmZzZXQiOjQ5NzU5NCwibl9vcGVuX3RhZ3MiOjAsImZnX2NvbG9yIjpudWxsLCJiZ19jb2xvciI6bnVsbCwic3R5bGVfbWFzayI6MCwic2VjdGlvbnMiOltdLCJsaW5lbm9faW5fc2VjdGlvbiI6MTJ9',
  append: false,
  truncated: false,
  offset: 0,
  size: 497594,
  total: 497594,

  lines: [
    // {
    //   offset: 1000,
    //   content: [{ text: 'Running with gitlab-runner 12.1.0 (de7731dd)' }],
    // },
    // {
    //   offset: 1001,
    //   content: [{ text: ' on docker-auto-scale-com 8a6210b8' }],
    // },
    // {
    //   offset: 1002,
    //   content: [
    //     {
    //       text:
    //         'Using Docker executor with image dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33',
    //     },
    //   ],
    //   section: 'prepare-executor',
    //   section_header: true,
    // },
    // {
    //   offset: 1003,
    //   content: [{ text: 'Starting service postgres:9.6.14 ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1004,
    //   content: [{ text: 'Pulling docker image postgres:9.6.14 ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1005,
    //   content: [
    //     {
    //       text: `Using docker image sha256:ac400042d32f8e146477ae5b3528f1e4232fc8e83b2fd9fe0cbe08900e45bae1 for
    //   postgres:9.6.14 ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1006,
    //   content: [{ text: 'Starting service redis:alpine ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1007,
    //   content: [{ text: 'Pulling docker image redis:alpine ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1008,
    //   content: [
    //     {
    //       text: `Using docker image sha256:ed7d2ff5a6232b43bdc89a2220ed989f532c3794422aa2a86823b8bc62e71447 for
    //   redis:alpine ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1009,
    //   content: [{ text: 'Waiting for services to be up and running...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1010,
    //   content: [
    //     {
    //       text: `Pulling docker image
    //   dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33
    //   ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1011,
    //   content: [
    //     {
    //       text: `Using docker image sha256:f5b71ccc7a53408ea88959433baf23f7d716bb431d0475a4f3d9ba213d7841c1 for
    //   dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33
    //   ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1012,
    //   content: [],
    //   section: 'prepare-executor',
    //   section_footer: true,
    //   section_duration: '10:00'
    // },
    // {
    //   offset: 1013,
    //   content: [
    //     {
    //       text: `Running on runner-8a6210b8-project-13083-concurrent-0 via
    //   runner-8a6210b8-gsrm-1566561403-37c881a0...`,
    //     },
    //   ],
    //   section: 'prepare-script',
    //   section_header: true,
    // },

    // {
    //   offset: 1014,
    //   content: [
    //     { text: 'Fetching changes with git depth set to 20..', style: 'term-fg-l-green term-bold' },
    //   ],
    //   section: 'get-sources',
    //   section_header: true,
    // },
    // {
    //   offset: 1015,
    //   content: [
    //     {
    //       text: 'Reinitialized existing Git repository in /builds/gitlab-org/gitlab-ce/.git/ ',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1016,
    //   content: [
    //     {
    //       text: 'From https://gitlab.com/gitlab-org/gitlab-ce',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1017,
    //   content: [
    //     {
    //       text: ' * [new branch] 66454-create-feature-flag -&gt; origin/66454-create-feature-flag',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1018,
    //   content: [
    //     {
    //       text: 'Checking out 061606b4 as 66454-create-feature-flag...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1019,
    //   content: [
    //     {
    //       text: 'Removing .gitlab_shell_secret',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1020,
    //   content: [
    //     {
    //       text: 'Removing .gitlab_workhorse_secret',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1021,
    //   content: [
    //     {
    //       text: 'Removing config/database_geo.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1022,
    //   content: [
    //     {
    //       text: 'Removing config/gitlab.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1023,
    //   content: [
    //     {
    //       text: 'Removing config/redis.cache.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1024,
    //   content: [
    //     {
    //       text: 'Skipping Git submodules setup',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1025,
    //   content: [],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1026,
    //   content: [],
    //   section: 'get-sources',
    //   section_footer: true,
    //   section_duration: '30:00'
    // },
    // {
    //   offset: 1027,
    //   content: [
    //     {
    //       text: 'Checking cache for debian-stretch-ruby-2.6.3-node-12.x-3...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    //   section_header: true,
    // },
    // {
    //   offset: 1028,
    //   content: [],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1029,
    //   content: [
    //     {
    //       text:
    //         'Downloading cache.zip from https://storage.googleapis.com/gitlab-com-runners-cache/project/13083/debian-stretch-ruby-2.6.3-node-12.x-3',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1030,
    //   content: [],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1031,
    //   content: [
    //     {
    //       text: 'Successfully extracted cache',
    //       style: 'term-fg-l-green term-bold ',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1032,
    //   content: [],
    //   sections: ['restore-cache'],
    //   section_footer: true,
    //   section_duration: '00:10'
    // },
    // {
    //   offset: 1033,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for compile-assets pull-cache (278912482)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    //   section_header: true,
    // },
    // {
    //   offset: 1034,
    //   content: [],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1035,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok ',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //     {
    //       text: ' id',
    //     },
    //     {
    //       text: '=278912482 responseStatus',
    //     },
    //     {
    //       text: '=200 OK token',
    //     },
    //     {
    //       text: '=DUNW1asp',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1036,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for setup-test-env (278912484)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1037,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok  id=278912484 responseStatus=200 OK token=gdMnE5zT',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1038,
    //   content: [
    //     {
    //       text: `WARNING: tmp/tests/gitlab-shell/.gitlab_shell_secret: chmod
    //       tmp/tests/gitlab-shell/.gitlab_shell_secret: no such file or directory (suppressing
    //       repeats)`,
    //       style: 'term-fg-yellow',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1039,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for retrieve-tests-metadata (278912485)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1040,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok id=278912485 responseStatus=200 OK token=rCtxUhsy',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1041,
    //   content: [],
    //   sections: ['download-artifacts'],
    //   section_footer: true,
    //   section_duration: '01:00:58'
    // },
    // {
    //   offset: 1042,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok ',
    //     },
    //     {
    //       text: ' id',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //     {
    //       text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
    //     },
    //   ],
    //   sections: 'download-artifacts',
    // },
    {
      offset: 1043,
      content: [{ text: 'Hello' }],
    },
    {
      offset: 5,
      content: [{ text: 'foo' }],
      section: 'prepare-script',
      section_header: true,
    },
    {
      offset: 52,
      content: [{ text: 'bar' }],
      section: 'prepare-script-nested',
      section_header: true,
    },
    {
      offset: 80,
      content: [{ text: 'this is a collapsible nested section' }],
      section: 'prepare-script-nested',
    },
    {
      offset: 90,
      section: 'prepare-script-nested',
      section_duration: '00:02',
    },
    {
      offset: 106,
      content: [],
      section: 'prepare-script',
      section_duration: '00:03',
    },
    {
      offset: 107,
      content: [],
      section_duration: '00:04',
    },
    {
      offset: 155,
      content: [],
    },
  ],
};
