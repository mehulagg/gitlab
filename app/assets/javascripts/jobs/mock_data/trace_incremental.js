// Using https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/278912562

export default {
  id: 278912562,
  status: 'failed',
  complete: true,
  state:
    'eyJvZmZzZXQiOjQ5NzU5NCwibl9vcGVuX3RhZ3MiOjAsImZnX2NvbG9yIjpudWxsLCJiZ19jb2xvciI6bnVsbCwic3R5bGVfbWFzayI6MCwic2VjdGlvbnMiOltdLCJsaW5lbm9faW5fc2VjdGlvbiI6MTJ9',
  append: false,
  truncated: false,
  offset: 0,
  size: 497594,
  total: 497594,

  lines: [
    {
      offset: 1042,
      content: [
        {
          text:'Not downloading anything ',
        },
        {
          text: ' id',
          style: 'term-fg-l-green term-bold',
        },
        {
          text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
        },
      ],
      sections: ['download-artifacts'],
    },

    {
      offset: 1043,
      content: [
        {
          text: 'A filipa Ã© fixe e o resto que se lixe ',
        },
        {
          text: ' id',
          style: 'term-fg-l-green term-bold',
        },
        {
          text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
        },
      ],
    },
  ],
};
