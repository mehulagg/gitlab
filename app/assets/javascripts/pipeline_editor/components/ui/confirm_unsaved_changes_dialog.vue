<script>
export default {
  props: {
    hasUnsavedChanges: {
      type: Boolean,
      required: true,
    },
  },
  created() {
    window.addEventListener('beforeunload', this.confirmChanges);
  },
  destroyed() {
    window.removeEventListener('beforeunload', this.confirmChanges);
  },
  methods: {
    confirmChanges(e) {
      if (this.hasUnsavedChanges) {
        e.preventDefault();
        e.returnValue = '';
      }
    },
  },
  render: () => null,
};
</script>
