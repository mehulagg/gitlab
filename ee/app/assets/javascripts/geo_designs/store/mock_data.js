import { __ } from '~/locale';

function generateRandomDateString() {
  return new Date(
    new Date(2019, 0, 1).getTime() +
      Math.random() * (new Date().getTime() - new Date(2019, 0, 1).getTime()),
  ).toString();
}

const mockData = [
  {
    id: 1,
    name: __("Zack's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'synced',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 2,
    name: __("Valery's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'pending',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 3,
    name: __("Mike's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'failed',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 4,
    name: __("Rachel's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: null,
    last_synced_at: null,
    last_verified_at: null,
    last_checked_at: null,
  },
];

export default mockData;
