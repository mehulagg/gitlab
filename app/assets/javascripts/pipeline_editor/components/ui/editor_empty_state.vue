<script>
import { GlSprintf } from '@gitlab/ui';
import { s__ } from '~/locale';

export default {
  components: {
    GlSprintf,
  },
  i18n: {
    title: s__('PipelineEditor|Optimize your workflow with CI/CD Pipelines'),
    body: s__(
      'PipelineEditor|Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
  },
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
  },
};
</script>
<template>
  <div class="gl-display-flex gl-flex-direction-column gl-align-items-center">
    <img :src="emptyStateIllustrationPath" data-testid="empty-state-illustration" />
    <h1 class="gl-font-size-h1">{{ $options.i18n.title }}</h1>
    <p>
      <gl-sprintf :message="$options.i18n.body" class="gl-display-inline">
        <template #code="{ content }">
          <code>{{ content }}</code>
        </template>
      </gl-sprintf>
    </p>
  </div>
</template>
