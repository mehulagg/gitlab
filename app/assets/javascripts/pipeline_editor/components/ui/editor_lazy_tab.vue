<script>
import { GlTab } from '@gitlab/ui';

/**
 * Wrapper of <gl-tab> to lazily render this tab's content
 * when shown **without dismounting it after**.
 *
 * Once the tab is selected it is permanently set as "not-lazy"
 * so it's contents are not dismounted.
 *
 * Usage:
 *
 * Same as <gl-tab>, for example:
 *
 * <gl-tabs>
 *   <editor-lazy-tab title="Tab 1"> ... </editor-lazy-tab>
 *   <editor-lazy-tab title="Tab 2"> ... </editor-lazy-tab>
 * </gl-tabs>
 */
export default {
  components: {
    GlTab,
  },
  props: {
    lazy: {
      type: Boolean,
      required: false,
      default: true,
    },
  },
  data() {
    return {
      isLazy: this.lazy,
    };
  },
  methods: {
    onClick() {
      this.isLazy = false;
    },
  },
};
</script>
<template>
  <gl-tab :lazy="isLazy" v-bind="$attrs" v-on="$listeners" @click="onClick">
    <slot v-for="slot in Object.keys($slots)" :slot="slot" :name="slot"></slot>
  </gl-tab>
</template>
