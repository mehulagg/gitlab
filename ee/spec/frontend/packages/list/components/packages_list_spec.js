import Vue from 'vue';
import { shallowMount } from '@vue/test-utils';
import { GlTable } from '@gitlab/ui';
import PackagesList from 'ee/packages/list/components/packages_list.vue';
import { packageList } from '../../mock_data';

describe('packages_list', () => {
  let wrapper;

  const findFirstActionColumn = (w = wrapper) => w.find({ ref: 'action-delete' });
  const findPackageListTable = (w = wrapper) => w.find({ ref: 'packageListTable' });
  const findPackageListSorting = (w = wrapper) => w.find({ ref: 'packageListSorting' });
  const findPackageListPagination = (w = wrapper) => w.find({ ref: 'packageListPagination' });
  const findPackageListDeleteModal = (w = wrapper) => w.find({ ref: 'packageListDeleteModal' });
  const findSortingItems = (w = wrapper) => w.findAll({ name: 'sorting-item-stub' });

  const defaultShallowMountOptions = {
    propsData: {
      canDestroyPackage: true,
    },
    stubs: {
      GlTable,
      GlSortingItem: { name: 'sorting-item-stub', template: '<div><slot></slot></div>' },
    },
    computed: {
      list: () => [...packageList],
    },
  };

  beforeEach(() => {
    // This is needed due to  console.error called by vue to emit a warning that stop the tests
    // see  https://github.com/vuejs/vue-test-utils/issues/532
    Vue.config.silent = true;
    wrapper = shallowMount(PackagesList, defaultShallowMountOptions);
  });

  afterEach(() => {
    Vue.config.silent = false;
    wrapper.destroy();
  });

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('contains a sorting component', () => {
    const sorting = findPackageListSorting();
    expect(sorting.exists()).toBe(true);
  });

  it('contains a table component', () => {
    const sorting = findPackageListTable();
    expect(sorting.exists()).toBe(true);
  });

  it('contains a pagination component', () => {
    const sorting = findPackageListPagination();
    expect(sorting.exists()).toBe(true);
  });
  it('contains a modal component', () => {
    const sorting = findPackageListDeleteModal();
    expect(sorting.exists()).toBe(true);
  });

  describe('when user can not destroy the package', () => {
    it('does not show the action column', () => {
      wrapper.setProps({ canDestroyPackage: false });
      const action = findFirstActionColumn();
      expect(action.exists()).toBe(false);
    });
  });

  describe('when the user can destroy the package', () => {
    it('show the action column', () => {
      const action = findFirstActionColumn();
      expect(action.exists()).toBe(true);
    });

    it('shows the correct deletePackageDescription', () => {
      expect(wrapper.vm.deletePackageDescription).toEqual('');

      wrapper.setData({ itemToBeDeleted: { name: 'foo' } });
      expect(wrapper.vm.deletePackageDescription).toEqual(
        'You are about to delete <b>foo</b>, this operation is irreversible, are you sure?',
      );
    });

    it('delete button set itemToBeDeleted and open the modal', () => {
      wrapper.vm.$refs.packageListDeleteModal.show = jest.fn();
      const [{ name, id }] = packageList.slice(-1);
      const action = findFirstActionColumn();
      action.vm.$emit('click');
      return Vue.nextTick().then(() => {
        expect(wrapper.vm.itemToBeDeleted).toEqual({ id, name });
        expect(wrapper.vm.$refs.packageListDeleteModal.show).toHaveBeenCalled();
      });
    });

    it('deleteItemConfirmation resets itemToBeDeleted', () => {
      wrapper.setData({ itemToBeDeleted: 1 });
      wrapper.vm.deleteItemConfirmation();
      expect(wrapper.vm.itemToBeDeleted).toEqual(null);
    });

    it('deleteItemCanceled resets itemToBeDeleted', () => {
      wrapper.setData({ itemToBeDeleted: 1 });
      wrapper.vm.deleteItemCanceled();
      expect(wrapper.vm.itemToBeDeleted).toEqual(null);
    });
  });

  describe('when the list is empty', () => {
    const findEmptySlot = (w = wrapper) => w.find({ name: 'empty-slot-stub' });

    beforeEach(() => {
      wrapper = shallowMount(PackagesList, {
        ...defaultShallowMountOptions,
        computed: { list: () => [] },
        slots: {
          'empty-state': { name: 'empty-slot-stub', template: '<div>bar</div>' },
        },
      });
    });

    it('show the empty slot', () => {
      const table = findPackageListTable();
      const emptySlot = findEmptySlot();
      expect(table.exists()).toBe(false);
      expect(emptySlot.exists()).toBe(true);
    });
  });

  describe('sorting component', () => {
    it('has all the sortable items', () => {
      const sortingItems = findSortingItems();
      expect(sortingItems.length).toEqual(wrapper.vm.sortableFields.length);
    });
  });
});
