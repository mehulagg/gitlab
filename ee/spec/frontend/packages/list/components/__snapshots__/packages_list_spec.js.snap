// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packages_list renders 1`] = `
<div
  class="d-flex flex-column align-items-end"
>
  <glsorting-stub
    class="my-3"
    isascending="true"
    sortdirectiontooltip="Sort direction"
    text="Name"
  >
    <div>
      
        Name
      
    </div>
    <div>
      
        Version
      
    </div>
    <div>
      
        Type
      
    </div>
    <div>
      
        Created
      
    </div>
  </glsorting-stub>
   
  <table
    aria-busy="false"
    aria-colcount="5"
    class="table b-table gl-table"
    id="__BVID__9"
  >
    <!---->
    <!---->
    <thead
      class=""
      role="rowgroup"
    >
      <!---->
      <tr
        role="row"
      >
        <th
          aria-colindex="1"
          class=""
          role="columnheader"
          scope="col"
        >
          Name
        </th>
        <th
          aria-colindex="2"
          class=""
          role="columnheader"
          scope="col"
        >
          Version
        </th>
        <th
          aria-colindex="3"
          class=""
          role="columnheader"
          scope="col"
        >
          Type
        </th>
        <th
          aria-colindex="4"
          class=""
          role="columnheader"
          scope="col"
        >
          Created
        </th>
        <th
          aria-colindex="5"
          aria-label="Actions"
          class=""
          role="columnheader"
          scope="col"
        >
          
        </th>
      </tr>
    </thead>
    <!---->
    <tbody
      class=""
      role="rowgroup"
    >
      <!---->
      <tr
        class=""
        role="row"
      >
        <td
          aria-colindex="1"
          class="w-25"
          role="cell"
        >
          <div
            class="flex-truncate-parent"
          >
            <a
              class="flex-truncate-child"
              data-qa-selector="package_link"
              href="/asd/lol"
            >
              
            Test package
          
            </a>
          </div>
        </td>
        <td
          aria-colindex="2"
          class=""
          role="cell"
        >
          
        1.0.0
      
        </td>
        <td
          aria-colindex="3"
          class=""
          role="cell"
        >
          
        maven
      
        </td>
        <td
          aria-colindex="4"
          class=""
          role="cell"
        >
          <timeagotooltip-stub
            cssclass=""
            time=""
            tooltipplacement="top"
          />
        </td>
        <td
          aria-colindex="5"
          class="text-right"
          role="cell"
        >
          <glbutton-stub
            aria-label="Remove package"
            title="Remove package"
            variant="danger"
          >
            <icon-stub
              name="remove"
              size="16"
            />
          </glbutton-stub>
        </td>
      </tr>
      <tr
        class=""
        role="row"
      >
        <td
          aria-colindex="1"
          class="w-25"
          role="cell"
        >
          <div
            class="flex-truncate-parent"
          >
            <a
              class="flex-truncate-child"
              data-qa-selector="package_link"
              href="/asd/lol"
            >
              
            @Test/package
          
            </a>
          </div>
        </td>
        <td
          aria-colindex="2"
          class=""
          role="cell"
        >
          
        
      
        </td>
        <td
          aria-colindex="3"
          class=""
          role="cell"
        >
          
        npm
      
        </td>
        <td
          aria-colindex="4"
          class=""
          role="cell"
        >
          <timeagotooltip-stub
            cssclass=""
            time=""
            tooltipplacement="top"
          />
        </td>
        <td
          aria-colindex="5"
          class="text-right"
          role="cell"
        >
          <glbutton-stub
            aria-label="Remove package"
            title="Remove package"
            variant="danger"
          >
            <icon-stub
              name="remove"
              size="16"
            />
          </glbutton-stub>
        </td>
      </tr>
      <!---->
      <!---->
    </tbody>
  </table>
   
  <glpagination-stub
    align="center"
    class="w-100"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnextpage="Go to next page"
    labelpage="function _default(page) {
        return \\"Go to page \\".concat(page);
      }"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next ›"
    perpage="20"
    prevtext="‹ Prev"
    totalitems="100"
    value="1"
  />
   
  <glmodal-stub
    modalclass=""
    modalid="confirm-delete-pacakge"
    ok-variant="danger"
    titletag="h4"
  >
      
    <p />
  </glmodal-stub>
</div>
`;
